File: tree-inorder-traversal/recur.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func inorderTraversal(root *TreeNode) []int {
    order := make([]int, 0, 25)
    inorderhelper(root, &order)
    return order
}

func inorderhelper(node *TreeNode, order *[]int) {
    if node == nil {
        return
    }
    
    inorderhelper(node.Left, order);
    *order = append(*order, node.Val)
    inorderhelper(node.Right, order)
}

File: two-sum-iv/recur.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def findTarget(self, root: Optional[TreeNode], k: int) -> bool:
        seen = set()
        return Solution.search(root, seen, k)

    def search(r: Optional[TreeNode], seen: set[int], k) -> bool:
        if r is None:
            return False
        if k - r.val in seen:
            return True
        seen.add(r.val)
        return Solution.search(r.left, seen, k) or Solution.search(r.right, seen, k)
